### PR TITLE
add FreeLabels and free labels on partial failure

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -633,13 +633,7 @@ int TSDB_alter(RedisModuleCtx *ctx, RedisModuleString **argv, int argc){
...
         RemoveIndexedMetric(ctx, keyName, series->labels, series->labelsCount);
         // free current labels
-        if (series->labelsCount > 0) {
-            for (int i = 0; i < series->labelsCount; i++) {
-                RedisModule_FreeString(ctx, series->labels[i].key);
-                RedisModule_FreeString(ctx, series->labels[i].value);
-            }
-            free(series->labels);
-        }
+       	FreeLabels(series->labels, series->labelsCount);
```
Comment (danni-m) [inline src/module.c]: @gkorland you have a useless tab here.

Comment (gkorland) [reply src/module.c]: removed

File: src/module.c [hunk 2]
```diff
@@ -50,7 +50,7 @@ static int parseLabelsFromArgs(RedisModuleString **argv, int argc, size_t *label
         	RedisModule_StringPtrLen(key, &keyLen);
         	RedisModule_StringPtrLen(value, &valueLen);
         	if(keyLen==0 || valueLen==0 || strpbrk( RedisModule_StringPtrLen(value, NULL), "(),")){
-        		free(labelsResult);
+        		FreeLabels(labelsResult, i); // need to release prior key values too
```
Comment (danni-m) [inline src/module.c]:
```suggestion
        		FreeLabels(labelsResult, i + 1); // need to release prior key values too
```

File: src/module.c [hunk 3]
```diff
@@ -50,7 +50,7 @@ static int parseLabelsFromArgs(RedisModuleString **argv, int argc, size_t *label
...
         		return REDISMODULE_ERR;
         	}
```
Comment (danni-m) [inline src/module.c]: @gkorland why not use the context here and depend on automemory?

File: src/module.c (hunk 2 above)
Comment (gkorland) [reply src/module.c]: why i+1?
The current round wasn't assigned yet
e.g. on the first round when i=0 I don't have anything to release

File: src/module.c (hunk 3 above)
Comment (danni-m) [reply src/module.c]: anyway, its not crtical for now, but we might want to consider all the parse function to use ctx to allow the automemory to clean on failure.

**Pull request merged**